tests/src/args.rs

```diff
@@ -1,19 +1,30 @@
+use regex::Regex;
+use std::path::PathBuf;
+
 use clap::{Parser, Subcommand};
 
 /// Typst's test runner.
 #[derive(Debug, Clone, Parser)]
+#[command(bin_name = "cargo test --workspace --test tests --")]
 #[clap(name = "typst-test", author)]
 pub struct CliArguments {
     /// The command to run.
     #[command(subcommand)]
     pub command: Option<Command>,
-    /// All the tests that contain the filter string will be run.
-    pub filter: Vec<String>,
-    /// Runs only the tests with the exact specified `filter` names.
+    /// All the tests whose names match the test name pattern will be run.
+    #[arg(value_parser = Regex::new)]
+    pub pattern: Vec<Regex>,
+    /// Restricts test selection within the given path.
+    #[arg(short, long, value_parser = |s: &str| PathBuf::from(s).canonicalize())]
```

Comment:
why canonicalize?

Reply:
canonicalize() checks the existence of the path (and reports error in the terminal) and converts it to an absolute path.